src/app/client/src/app/app.component.ts

```diff
@@ -122,7 +123,42 @@ export class AppComponent implements OnInit, OnDestroy {
           _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideHeaderNFooter');
       });
   }
+  public makeUTMSession(params) {
+    for (const item in params) {
+      if (params.hasOwnProperty(item)) {
+        switch (item) {
+          case ('channel'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case ('utm_campaign'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case 'utm_medium':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmMedium' });
+            break;
+          case 'utm_source':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmSource' });
+            break;
+          case 'utm_term':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmTerm' });
+            break;
+          case 'utm_content':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmContent' });
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    sessionStorage.setItem('UTM', JSON.stringify(this.resultJson));
+  }
   ngOnInit() {
+    this.activatedRoute.queryParams.subscribe(params => {
+      const utmParams = ['utm_campaign', 'utm_medium', 'utm_source', 'utm_term', 'utm_content', 'channel'];
```

Comment:
Am not sure we will get queryParam of the chiled route in rootComponent. Can you verify this

Reply:
Yes verified it's not only for root if we pass the parameter in any route from any place I am capturing that and processing as per requirement.